### DISCUSSION TITLE
encoding ?

### DISCUSSION THREAD

**Discussion by JurjendeGroot** [QuestPDF/QuestPDF#648] [Q&A]
I'm experiencing problem with characters not being display well in the resulting PDF.

'Bucureşti' in the datasource of the document is being shown as 'Bucure?ti' in the resulting PDF. (see attachments)

I've searched the QuestPdf website (and via google) but could not find a solution.
Help would be greatly appreciated.

**Comment by MarcinZiabek** [comment]
Hi 😀 

Would you please share more information about the font used? Did you change this font in any way (e.g., glyph subsetting)? Having a minimal example may help during debugging.

**Reply by JurjendeGroot** [reply]
Hi,

Thanks for responding to my question. :-)

Other than setting the fontColor or FontSize I'm not doing anything with fonts.
I've attached a simple example project for you showing the issue.

[questPdfCharacterEncodingIssue.zip](https://github.com/QuestPDF/QuestPDF/files/12682375/questPdfCharacterEncodingIssue.zip)

**Reply by MarcinZiabek** [reply]
I am sorry for the late response 😀

Once I have changed the following configuration in your code `QuestPDF.Settings.CheckIfAllTextGlyphsAreAvailable = true;`, I got a proper exception. Its message specifies that your character is not available in the chosen font. You need to change the target font to another one or use the font fallback functionality.

Also, the exception message specifies which fonts in your local environment do support the missing glyph.

<img width="1950" alt="image" src="https://github.com/QuestPDF/QuestPDF/assets/9263853/bcfeca77-f2d2-49d3-a7c4-492da9750bad">

Example:

<img width="808" alt="image" src="https://github.com/QuestPDF/QuestPDF/assets/9263853/2ea40b0f-f8d7-4bc8-8e70-9cfa14837f85">

Results with:
<img width="564" alt="image" src="https://github.com/QuestPDF/QuestPDF/assets/9263853/99b54dde-630e-48bb-8dfb-dc5aecc49e8c">

**Reply by MarcinZiabek** [reply]
If you have any suggestions on how we can improve the developer experience and better communicate this issue, please share your feedback 😀

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
